crypto/entropy_test.go

```diff
@@ -0,0 +1,40 @@
+package crypto_test
+
+import (
+	"testing"
+
+	"github.com/alecthomas/assert/v2"
+	"github.com/tbd54566975/web5-go/crypto"
+)
+
+func Test_GenerateEntropy(t *testing.T) {
+	size := 16
+	bytes, err := crypto.GenerateEntropy(size)
+	assert.NoError(t, err)
+	assert.Equal(t, size, len(bytes))
+}
+
+func Test_GenerateEntropy_InvalidSize(t *testing.T) {
+	bytes, err := crypto.GenerateEntropy(0)
+	assert.Error(t, err)
+	assert.Equal(t, nil, bytes)
+
+	bytes, err = crypto.GenerateEntropy(-1)
+	assert.Error(t, err)
+	assert.Equal(t, nil, bytes)
+}
+
+func Test_GenerateHexNonce(t *testing.T) {
+	nonceMap := make(map[string]bool)
+
+	for i := 0; i < 1000000; i++ {
```

Comment:
is 1 million excessive? do we even want a test like this or should we just test one and call it a day?

Reply:
It's excessive, though I appreciate the idea. We're depending on the randomness of the `rand.Read` implementation, and we should trust go's implementation. It's only useful for us to test the conditions that we're adding ourselves. So, I would test the success case that:
* The length of the byte slice is 16
* Error is nil
* Maybe try hex decoding it to validate that it's hex. I'm on the fence if that's useful.

I'm usually a testing purist who wants to cover all code paths, but I'm not sure how to simulate error cases in a realistic or useful way. Maybe testing the success case is all we need.

Reply:
Agreed, alright cool

I like the hex decoding, tho it is slightly awkward. The reason I like it is because the test hard-codes the assumption the length of the string is 32 characters, even tho that abstraction is encapsulated within the function. In the docs comment I call out both the length and it being hex-encoded. So yeah, I think it's sage to assume the nonce should be locked-in as hex-encoded.

Updated!